### ISSUE TITLE
BUG: pandas.groupby([A, B]).sum(level='B') gives cython_agg_general() error

### ISSUE THREAD

**Issue by kahyforest** [pandas-dev/pandas#34235]
#### Code Sample, a copy-pastable example

```python
import pandas as pd

A = [1, 2, 3]
B = [5, 5, 5]
C = [3, 2, 1]

df = pd.DataFrame(data={'A': A, 'B': B, 'C': C})
print("Method 1:\n", df.groupby(['A', 'C']).agg({'C': 'sum'}))  # works
print("Method 2:\n", df.groupby(['A', 'C']).sum(level='C'))  # gives error
```

#### Problem description

Last time I ran my code I could use the keyword _level_ to do statistics over a particular multiindex. Now, I get an error. Using agg is a workaround, but I'll have to rewrite quite a lot of code then. I couldn't find anything about the level kwarg being deprecated. Current output:

> Traceback (most recent call last):
> 
>   File "C:\Users\kahy\OneDrive - Skogforsk\Documents\Projekt\Översikt virkesmätning\Code\bugtest.py", line 19, in <module>
>     print("Method 2:\n", df.groupby(['A', 'C']).sum(level='C'))  # gives error
> 
>   File "C:\Program Files\Miniconda3\envs\kari_env\lib\site-packages\pandas\core\groupby\groupby.py", line 1371, in f
>     return self._cython_agg_general(alias, alt=npfunc, **kwargs)
> 
> TypeError: _cython_agg_general() got an unexpected keyword argument 'level'
> #### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.2.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en
LOCALE           : Swedish_Sweden.1252

pandas           : 1.0.3
numpy            : 1.18.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.2.0.post20200511
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : 3.0.3
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.13.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None
numba            : None
None

</details>

**Comment by ldrobner** [comment]
take

**Comment by ldrobner** [comment]
@charlesdong1991 @kahyforest @anyone_who_can_help

I have been digging around looking for the problem, here are a couple items I noticed:

1. I think the problem is in the decorator method in `.\pandas\util\_decorators.py`. I believe this is the decorator function you use for a lot of functions, and would probably be hard to change. I looked around to find some similar errors and solutions like this one - https://github.com/dask/dask/issues/5029 - where they use error checking to properly decorate a callable.

2. Since it seems like a problem with the sum() function's allowable keywords I believe that the error reported above is misleading and probably needs to be fixed. The error reported above is raised in mean() in `.\pandas\core\groupby\groupby.py` even though we call sum() to get the error in this case. I might be wrong about the trace, but the error seems wrong to me.

---

I want to discuss a solution. I haven't spent too much time digging to find a better place to fix this error. In my mind we can: 
* Change the decorator function I mentioned, but I think that is a bad solution. 
* Change the keywords we can pass to sum()? 
   * Or maybe we can check for a bunch of keywords and reduce or replace the keywords based on valid/invalid keywords.
   * I also have no clue where to do this

---
I just started poking around so let me know if any of this makes sense. I would love to hear your thoughts and opinions. 

Thanks
